src/login/logind-core.c

```diff
@@ -734,8 +742,8 @@ int manager_read_utmp(Manager *m) {
                 if (isempty(t))
                         continue;
 
-                s = hashmap_get(m->sessions_by_leader, PID_TO_PTR(u->ut_pid));
-                if (!s)
+                r = manager_get_session_by_pidref(m, &PIDREF_MAKE_FROM_PID(u->ut_pid), &s);
+                if (r <= 0)
```

Comment:
`r` is not used later, hence the lines can be shorten by `if (manager_get_session_by_pidref(...) <= 0)`.
Feel free to ignore you do not like the form.